chrome/content/zotero/zoteroPane.js

```diff
@@ -6626,6 +6596,34 @@ var ZoteroPane = new function()
 		}
 	};
 
+	this.showFileRenamingBanner = function () {
+		if (Zotero.Prefs.get('autoRenameFiles.bannerShown')) {
+			return;
+		}
+		if (Zotero.Prefs.get('autoRenameFiles.done')) {
+			// If user had a custom `attachmentRenameTemplate` (now a synced
+			// setting), `autoRenameFiles.done` was set to `false` during
+			// migration. Otherwise, it's `true` (the default), and no banner
+			// is needed.
```

Comment:
Wait, why isn't it needed just because they didn't have a custom format? We want to show this for all upgrading users who have automatic renaming on, because we want them to know that filenames will now be changing automatically even after they're first added.

`bannerShown` also isn't a default pref, so even once `true` it will be saved forever in people's `prefs.js`, and clearing the prefs will cause it to appear again.

 Seems like we should make `bannerShown` a default pref, set it `true`, and then in pref migration set it to `false` if `autoRenameFiles` is set to true. We can then remove this block and just show when `bannerShown` is `false`.

Reply:
You're right, that makes much more sense, fixed!